### PR TITLE
TESB416-243: ZKM Scraper (including MovieDB changes)

### PR DESCRIPTION
**Pull request by rgcjonas**

### CODE REVIEW

**Comment by rgcjonas** [comment]
@flitzpiepe96 @W-i-Z-o I got the tests fixed you may start the review now

File: commons/build.gradle [hunk 1]
```diff
@@ -28,6 +28,6 @@ repositories {
 }
 
 dependencies {
-    compile('org.hibernate.javax.persistence:hibernate-jpa-2.1-api:1.0.0.Final')
+    compile('org.springframework.boot:spring-boot-starter-data-jpa:1.5.3.RELEASE')
```
Comment (flitzpiepe96) [inline commons/build.gradle]: Why do we need spring here? Commons should not know anything about spring, Hibernate should be enough here.

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KurbelScraper.java [hunk 2]
```diff
@@ -45,7 +47,12 @@ private void handleMovies() {
         for (WebElement movieElement : movies) {
             String title = movieElement.findElement(By.xpath(".//h2")).getText();
             Movie movie = saveObject(new Movie(title, null, null, 0, null, null), Movie.class);
-            retrieveMovieInformation(movie);
+            try {
+                movie = retrieveMovieInformation(movie);
+            } catch (MovieDbException e) {
+                getLogger().warn("Failed to retrieve data from movie db for " + title);
+                getLogger().warn(e.toString());
```
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KurbelScraper.java]: Make logger in superclass protected and use that one. We dont want to call LoggerFactory (and possibly create a new logger) each time.

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        List<MediaCreditCrew> list = crew.stream()
+                .filter(c -> c.getDepartment().equals("Writing") && c.getJob().equals("Screenplay"))
+                .collect(Collectors.toList());
+
+        if (list.size() > 0) {
+            Iterator<MediaCreditCrew> i = list.iterator();
+            while (i.hasNext()) {
+                MediaCreditCrew item = i.next();
+                result += item.getName() + (i.hasNext() ? ", " : "");
+            }
```
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java]: Could we extract something like a "concatStrings" method here? This looks like a lot of duplicate code across the whole file

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    // char[] chars = new char[1024];
+    // while((read = reader.read(chars)) != -1){
+    // buffer.append(chars, 0, read);
+    // }
+    // return buffer.toString();
+    // } finally {
+    // if (reader != null){
+    // reader.close();
+    // }
+    // }
```
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java]: Why is this commented out? Just delete it if its not needed, otherwise add some comment why we need to keep it

File: web app/src/main/java/de/tinf15b4/kino/web/rest/RestClient.java [hunk 5]
```diff
@@ -139,7 +139,8 @@ public RestResponse getRatedCinemas(long cinemaId) {
     }
 
     public RestResponse getPlaylistForCinemas(long cinemaId, Date from, Date to) {
-        String requestUrl = baseUrl + String.format(GET_PLAYLIST_CINEMA, cinemaId, from.getTime(), to.getTime());
+        String requestUrl = baseUrl + String.format(GET_PLAYLIST_CINEMA, cinemaId, (from != null) ? from.getTime() : "",
+                (to != null) ? to.getTime() : "");
```
Comment (flitzpiepe96) [inline web app/src/main/java/de/tinf15b4/kino/web/rest/RestClient.java]: Add some tests for the null cases

File: web app/src/main/java/de/tinf15b4/kino/web/views/MovieView.java [hunk 6]
```diff
@@ -92,7 +92,8 @@ public void enter(ViewChangeEvent event) {
                     }
 
                     RestResponse playlistResponse = userBean.getRestClient().getPlaylistForMovie(m.getId(), new Date(),
-                            new Date(new Date().getTime() + 1000L * 3600 * 24 * 7));
+                            null);
+//                            new Date(new Date().getTime() + 1000L * 3600 * 24 * 7));
```
Comment (flitzpiepe96) [inline web app/src/main/java/de/tinf15b4/kino/web/views/MovieView.java]: remove the comment line

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java (hunk 3 above)
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/tmdb/TmdbDataRetriever.java]: I think String.join might be appropriate here
